packages/core/content-releases/server/src/bootstrap.ts

```diff
@@ -1,27 +1,42 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
-import type { LoadedStrapi, Entity as StrapiEntity } from '@strapi/types';
+import type { Common, LoadedStrapi, Entity as StrapiEntity } from '@strapi/types';
 
-import { RELEASE_ACTION_MODEL_UID } from './constants';
-import { getService } from './utils';
+import { RELEASE_ACTION_MODEL_UID, RELEASE_MODEL_UID } from './constants';
+import { getEntryValidStatus, getService } from './utils';
 
 const { features } = require('@strapi/strapi/dist/utils/ee');
 
 export const bootstrap = async ({ strapi }: { strapi: LoadedStrapi }) => {
   if (features.isEnabled('cms-content-releases')) {
     // Clean up release-actions when an entry is deleted
     strapi.db.lifecycles.subscribe({
-      afterDelete(event) {
+      async afterDelete(event) {
         // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
         const { model, result } = event;
         // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
         if (model.kind === 'collectionType' && model.options?.draftAndPublish) {
           const { id } = result;
-          strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
+
+          const releases = await strapi.db.query(RELEASE_MODEL_UID).findMany({
+            where: {
+              actions: {
+                target_type: model.uid,
+                target_id: id,
+              },
+            },
+          });
+
+          await strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
             where: {
               target_type: model.uid,
               target_id: id,
             },
           });
+
+          // We update the status of each release after delete the actions
+          for (const release of releases) {
+            getService('release', { strapi }).updateReleaseStatus(release.id);
+          }
```

Comment:
What happens if the update fails on one of the releases?

Reply:
In that case, and considering this is part of the edit flow, it would throw an error when you try to save the entry.

Reply:
Does that mean you are not able to save the entry if it fails to update the release?

Reply:
Yes, Do you think we should add a try ... catch here to handle the error?